### DISCUSSION TITLE
@apply does not work in styled-jsx at vite configuration.

### DISCUSSION THREAD

**Discussion by airman5573** [tailwindlabs/tailwindcss#12729] [Help]
```typescript
// vite.config.ts

import react from "@vitejs/plugin-react";
import tsconfigPaths from "vite-tsconfig-paths";
import { defineConfig } from "vite";
import { resolve } from "path";

// https://vitejs.dev/config/
export default defineConfig({
    build: {
        ...
    },
    plugins: [
        react({
            babel: {
                plugins: ["styled-jsx/babel"],
            },
        }),
        tsconfigPaths(),
    ],
    publicDir: false,
});
```

```javascript
// postcss.config.js

module.exports = {
    plugins: {
        tailwindcss: {},
        autoprefixer: {},
        "postcss-prefixwrap": ".oico-root",
    },
};
```

And this code does not work.

```javascript
<style jsx>{`
    .checkbox-group-wrapper :global(.om-checkbox label) {
        @apply text-240;
    }
`}</style>
```

<img width="390" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/68623798/1193c54c-ff0c-4b3f-9e87-240a784f1184">

How to make `@apply` work in this situation ?

**Comment by AM-NiceDay** [comment]
Hello @airman5573!

The `@apply` directive is a feature of Tailwind CSS that allows you to apply utility classes in CSS blocks. However, when using `styled-jsx`, a CSS-in-JS library for styling in a Next.js project, you might encounter issues because `styled-jsx` does not natively support Tailwind's `@apply` directive.

Here's what's likely going wrong and how to get around it:

1. **Incompatibility**: `styled-jsx` expects regular CSS and may not understand Tailwind-specific syntax like `@apply`.

2. **Babel Configuration**: Even with the right Babel plugins, `styled-jsx` might not be set up to work with the Tailwind `@apply` directive.

To work around this:

- **Extract Classes**: Instead of using `@apply` within `styled-jsx`, extract the classes you want to apply and use them directly in your JSX as className values.

- **Custom Babel Plugin**: You may use a custom Babel plugin that can transform `@apply` directives into a format that `styled-jsx` can understand. I found this one, which is reported to help with the issue some time ago, but is not actively maintained anymore - [https://github.com/giuseppeg/styled-jsx-plugin-postcss](https://github.com/giuseppeg/styled-jsx-plugin-postcss). More on this here - [@apply doesn't work inside <style jsx></style>](https://github.com/tailwindlabs/tailwindcss/issues/1234#issuecomment-557500287).

- **Sticking to the Tailwind CSS**: Anyway, when working with Tailwind CSS, you may want to abandon CSS-in-JS and use Tailwind exclusively. This is how it is supposed to be used in the first place.
